Exec/radiation_tests/RadThermalWave/Problem_Derive.cpp

```diff
@@ -4,56 +4,117 @@
 #include <Problem_Derive_F.H>
 #include <Castro.H>
 #include <Castro_F.H>
+#include <fundamental_constants.H>
+#include <prob_parameters.H>
+#include <cmath>
 
 using namespace amrex;
 
-#ifdef __cplusplus
-extern "C"
+
+void dertexact(const Box& bx, FArrayBox& derfab, int dcomp, int /*ncomp*/,
+               const FArrayBox& datfab, const Geometry& geomdata,
+               Real time, const int* /*bcrec*/, int /*level*/)
+{
+
+  const auto dx = geomdata.CellSizeArray();
+  const auto problo = geomdata.ProbLoArray();
+
+  auto const dat = datfab.array();
+  auto const der = derfab.array();
+
+  amrex::ParallelFor(bx,
+  [=] AMREX_GPU_HOST_DEVICE (int i, int j, int k) noexcept
+  {
+
+    Real loc[3];
+
+    loc[0] = problo[0] + (static_cast<Real>(i) + 0.5_rt) * dx[0];
+    loc[1] = problo[1] + (static_cast<Real>(j) + 0.5_rt) * dx[1];
```

Comment:
I think we need to guard this against invalid accesses in 1D and 2D.

Reply:
fixed (in 2 places)